File: common/src/main/java/juuxel/adorn/block/entity/KitchenCupboardBlockEntity.java
package juuxel.adorn.block.entity;

import juuxel.adorn.block.AdornBlockEntities;
import juuxel.adorn.menu.KitchenCupboardMenu;
import juuxel.adorn.util.AdornUtil;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.entity.player.Inventory;
import net.minecraft.world.inventory.AbstractContainerMenu;
import net.minecraft.core.BlockPos;

public final class KitchenCupboardBlockEntity extends SimpleContainerBlockEntity {
    public KitchenCupboardBlockEntity(BlockPos pos, BlockState state) {
        super(AdornBlockEntities.KITCHEN_CUPBOARD.get(), pos, state, 15);
    }

    @Override
    protected AbstractContainerMenu createMenu(int syncId, Inventory inv) {
        return new KitchenCupboardMenu(syncId, inv, this, AdornUtil.menuContextOf(this));
    }
}

File: common/src/main/kotlin/juuxel/adorn/block/entity/KitchenCupboardBlockEntity.kt
package juuxel.adorn.block.entity

import juuxel.adorn.block.AdornBlockEntities
import juuxel.adorn.menu.KitchenCupboardMenu
import juuxel.adorn.util.menuContextOf
import net.minecraft.block.BlockState
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.util.math.BlockPos

class KitchenCupboardBlockEntity(pos: BlockPos, state: BlockState) :
    SimpleContainerBlockEntity(AdornBlockEntities.KITCHEN_CUPBOARD, pos, state, 15) {
    override fun createMenu(syncId: Int, playerInv: PlayerInventory) =
        KitchenCupboardMenu(syncId, playerInv, this, menuContextOf(this))
}
